azurelinuxagent/ga/monitor.py

```diff
@@ -277,8 +281,25 @@ def daemon(self):
             self.send_host_plugin_heartbeat()
             self.send_imds_heartbeat()
             self.log_altered_network_configuration()
+            self.reset_loggers()
             time.sleep(min_delta)
 
+    def reset_loggers(self):
+        """
+        The loggers maintain hash-tables in memory and they need to be cleaned up from time to time.
+        For reference, please check azurelinuxagent.common.logger.Logger and
+        azurelinuxagent.common.event.EventLogger classes
+        """
+        time_now = datetime.datetime.utcnow()
+        if not self.last_reset_loggers_time:
+            self.last_reset_loggers_time = time_now
+
+        if time_now >= (self.last_reset_loggers_time +
+                        MonitorHandler.RESET_LOGGERS_PERIOD):
+            logger.reset_periodic()
```

Comment:
I know this is the pattern we use today for all these periodic actions; we should consider if instead we should follow this:
```
    try:
        logger.reset_periodic()
   finally:
         self.last_reset_loggers_time = time_now
```
so that if there is an exception we don't keep calling the reset code continuously (to do after release)

Reply:
+1